lib/screens/profile_screens/edit_name/edit_name_screen.dart

```diff
@@ -1,94 +1,74 @@
 import 'package:seeds/components/text_form_field_custom.dart';
 import 'package:seeds/datasource/remote/model/profile_model.dart';
+import 'package:seeds/domain-shared/event_bus/event_bus.dart';
+import 'package:seeds/domain-shared/event_bus/events.dart';
+import 'package:seeds/domain-shared/page_command.dart';
 import 'package:seeds/domain-shared/page_state.dart';
 import 'package:seeds/domain-shared/ui_constants.dart';
 import 'package:seeds/screens/profile_screens/edit_name/interactor/viewmodels/edit_name_bloc.dart';
+import 'package:seeds/screens/profile_screens/edit_name/interactor/viewmodels/page_commands.dart';
 import 'package:seeds/utils/build_context_extension.dart';
 
-class EditNameScreen extends StatefulWidget {
+class EditNameScreen extends StatelessWidget {
   const EditNameScreen({Key? key}) : super(key: key);
 
-  @override
-  _EditNameScreenState createState() => _EditNameScreenState();
-}
-
-class _EditNameScreenState extends State<EditNameScreen> {
-  late EditNameBloc _editNameBloc;
-  ProfileModel? _profileModel;
-  final _nameController = TextEditingController();
-  final _formKeyName = GlobalKey<FormState>();
-
-  @override
-  void initState() {
-    super.initState();
-    _editNameBloc = EditNameBloc();
-    _nameController.addListener(() {
-      _editNameBloc.add(OnNameChanged(name: _nameController.text));
-    });
-  }
-
-  @override
-  void dispose() {
-    _nameController.dispose();
-    super.dispose();
-  }
-
-  void _onSubmitted() {
-    if (_formKeyName.currentState!.validate()) {
-      _editNameBloc.add(SubmitName(profile: _profileModel));
-    }
-  }
-
   @override
   Widget build(BuildContext context) {
-    _profileModel = ModalRoute.of(context)!.settings.arguments as ProfileModel?;
-    _nameController.text = _profileModel?.nickname ?? '';
+    final _profileModel = ModalRoute.of(context)!.settings.arguments as ProfileModel?;
+    assert(_profileModel != null);
     return Scaffold(
       appBar: AppBar(title: Text(context.loc.editNameTitle)),
       body: BlocProvider(
-        create: (_) => _editNameBloc,
+        create: (_) => EditNameBloc(_profileModel!),
         child: BlocConsumer<EditNameBloc, EditNameState>(
-          listenWhen: (previous, current) =>
-              previous.pageState != PageState.success && current.pageState == PageState.success,
-          listener: (context, state) => Navigator.of(context).pop(state.name),
-          builder: (_, state) {
+          listenWhen: (_, current) => current.pageCommand != null,
+          listener: (context, state) {
+            final pageCommand = state.pageCommand;
+
+            if (pageCommand is EditNameSuccess) {
+              Navigator.of(context).pop(state.name);
+            } else if(pageCommand is ShowErrorMessage) {
+              eventBus.fire(ShowSnackBar(pageCommand.message));
+            }
+            BlocProvider.of<EditNameBloc>(context).add(const ClearPageCommand());
```

Comment:
I think the clear event only applies insiide ShowErrorMessage because on EditNameSuccess the screen is poped

Reply:
agree done